src/Utils/Image.php

```diff
@@ -639,6 +653,7 @@ public function send(int $type = self::JPEG, ?int $quality = null): void
 	/**
 	 * Outputs image to browser or file.
 	 * @throws ImageException
+	 * @param self::JPEG|self::PNG|self::GIF|self::WEBP|self::AVIF|self::BMP $type
```

Comment:
This type is repeated quite a lot, maybe type alias would be useful? https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases

Reply:
Yes, but I wanted to stay away from phpstan specific annotations, if possible.

Another alternative would be `key-of<self::Formats>`, but I'm not sure how well that is supported in the broader tooling.

Reply:
Psalm and phpstan suppport each others annotations. Also phpstorm adds support for new phpstan and psalm annotations with each version.

Others probably burn when anything advanced is used...

Reply:
(I'm thinking that in nette/utils 4.1, which will require newer PHP, it can be replaced somehow with enum.)

Reply:
If you want class now, use enum-like class. Changing it safely to enum needs changing constants to cases, removing compatibility methods and deprecating static constructors
https://github.com/orisai/auth/blob/9e35c56fdab1871bdb903b693441574e23fe0b2a/src/Authentication/LogoutCode.php

Reply:
It could perhaps be prearranged like this fe69e377b33862e73f2859beea95deb5c49cfa90